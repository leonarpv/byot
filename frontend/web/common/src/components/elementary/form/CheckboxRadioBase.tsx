import * as React from 'react';
import {FormControlLabel, FormControlLabelProps, makeStyles, Theme, Typography} from '@material-ui/core';
import {ClassesObject, CombineClasses} from '../../../types/CombineClasses';
import {WithStyles} from '../../../types/WithStyles';

export type Props<T extends ClassesObject<C>, C = any> = T &
  CombineClasses<WithStyles<typeof styles>, T> & {
    label: string;
    component: React.ComponentType<T>;
    FormControlLabelProps?: Omit<FormControlLabelProps, 'control' | 'label'>;
  };

const styles = (theme: Theme) => ({
  label: {},
});
const useStyles = makeStyles(styles);

export const CheckboxRadioBase = <T extends ClassesObject<C>, C = any>(props: Props<T, C>) => {
  const styles = useStyles(props);
  const {component, ...radioProps} = props;
  return (
    <FormControlLabel
      {...props.FormControlLabelProps}
      control={<props.component {...(radioProps as any)} />}
      label={
        ['string', 'number'].includes(typeof props.label) ? (
          <Typography classes={{root: styles.label}}>{props.label}</Typography>
        ) : (
          props.label
        )
      }
    />
  );
};
