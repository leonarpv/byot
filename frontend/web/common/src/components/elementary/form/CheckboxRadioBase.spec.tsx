import * as React from 'react';
import {render} from '@testing-library/react';
import {CheckboxRadioBase} from './CheckboxRadioBase';

describe('<CheckboxRadioBase/>', () => {
  it('should render', () => {
    const {container} = render(<CheckboxRadioBase label="Label" component={() => <div />} />);
    expect(container).toMatchSnapshot();
  });
});
