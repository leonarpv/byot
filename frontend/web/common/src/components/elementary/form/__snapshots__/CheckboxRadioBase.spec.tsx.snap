// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxRadioBase/> should render 1`] = `
<div>
  <label
    class="MuiFormControlLabel-root"
  >
    <div />
    <span
      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Label
      </p>
    </span>
  </label>
</div>
`;
